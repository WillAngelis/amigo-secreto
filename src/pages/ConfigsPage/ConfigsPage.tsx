import { Card } from '../../components/Card/Card';
import { Footer } from '../../components/Footer/Footer';
import { Form } from '../../components/Form/Form';
import { ListaDeParticipantes } from '../../components/ListaDeParticipantes/ListaDeParticipantes';

export const ConfigsPage = () => {
  return (
    <Card>
      <section>
        <h2>Vamos come√ßar!</h2>
        <Form />
        <ListaDeParticipantes />
        <Footer />
      </section>
    </Card>
  );
};
