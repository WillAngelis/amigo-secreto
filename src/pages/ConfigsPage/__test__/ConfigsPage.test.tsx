import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import { ConfigsPage } from '../ConfigsPage';

const mockNavegacao = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavegacao,
  };
});

describe('a pagina de configuracao', () => {
  test('deve ser renderizada corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <ConfigsPage />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
