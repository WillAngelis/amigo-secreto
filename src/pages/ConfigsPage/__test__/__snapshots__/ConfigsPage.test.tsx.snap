// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuracao deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Vamos come√ßar!
      </h2>
      <form>
        <div
          class="group-input-btn"
        >
          <input
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
            type="submit"
          >
            Adicionar
          </button>
        </div>
      </form>
      <ul />
      <footer
        class="footer-configs"
      >
        <button
          class="btn"
          disabled=""
        >
          Iniciar brincadeira
        </button>
        <img
          alt="Sacolas de compras"
          src="/imgs/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
